cmd/az-analyze/cmd/azva.go

```diff
@@ -0,0 +1,301 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"os"
+	v1beta1 "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/azuredisk/v1beta1"
+	azDiskClientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned"
+	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
+	"strings"
+	"time"
+)
+
+// azvaCmd represents the azva command
+var azvaCmd = &cobra.Command{
+	Use:   "azva",
+	Short: "Azure Volume Attachment",
+	Long:  `Azure Volume Attachment is a Kubernetes Custom Resource.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		pod, _ := cmd.Flags().GetString("pod")
+		node, _ := cmd.Flags().GetString("node")
+		zone, _ := cmd.Flags().GetString("zone")
+		namespace, _ := cmd.Flags().GetString("namespace")
+
+		numFlag := cmd.Flags().NFlag()
+		if hasNamespace := namespace != ""; hasNamespace {
+			numFlag--
+		}
+
+		var result []AzvaResource
+		// access to config and Clientsets
+		config := getConfig()
+		clientsetK8s := getKubernetesClientset(config)
+		clientsetAzDisk := getAzDiskClientset(config)
+
+		if numFlag > 1 {
+			fmt.Printf("only one of the flags is allowed.\n" + "Run 'az-analyze --help' for usage.\n")
+		} else {
+			if numFlag == 0 {
+				// if no flag value is provided , list all of the pods/nodes/zone information
+				result = GetAzVolumeAttachementsByPod(clientsetK8s, clientsetAzDisk, pod, namespace)
```

Comment:
When no flags are provided, it would be better if instead of rendering 3 tables, we render a combined table that contains Pod,Node and Zone as columns.

Reply:
It could be. Since the headers of first column are different; that is, PODNAME, NODENAME, ZONENAME. Do you want to keep header of each table or just have a sharing one? If the latter, we need a common name for three tables.

Reply:
I think we should have a single table with 3 columns and these 3 columns should have header names as: PODNAME, NODENAME, ZONENAME.

Reply:
created new func GetAllAzVolumeAttachements to handle this.